### PR TITLE
Fix optional arg typing in napari._qt.containers

### PR DESCRIPTION
**Pull request by dstansby**
# Fixes/Closes
More work towards https://github.com/napari/napari/issues/2751

# Description
This PR fixes instances of arguments in `napari._qt.containers` that should have `Optional` typing.

### CODE REVIEW

**Review by Carreau — APPROVED** [review]

**Review by andy-sweet — APPROVED** [review]
This one looks nice and straightforward to me. Will merge this after 24 hours unless there are any objections.

**Comment by andy-sweet** [comment]
@dstansby : FYI I updated the first comment / description before squashing and merging because we use that for the merged commit message on `main`. I removed most of the empty template stuff since that's mostly for an initial checklist or for new contributors, so feel free to remove/edit those in the future as you see fit.

**Pull request merged**
